File: WeirdAlgorithm.java
import java.util.Scanner;

class WeirdAlgorithm {

    private static void solve(long n) {

        while (true) {
            System.out.print(n + " ");
            if (n == 1) {
                break;
            } else if (n %2 == 1) {
                n = 3*n + 1;
            } else {
                n = n / 2;
            }
        }
    }

    public static void main(String[] args) {
        
        Scanner scanner = new Scanner(System.in);
        long n = scanner.nextLong();
        solve(n);
        scanner.close();
    }
}

File: weird_algorithm.py
def weird_algorithm(n: int):
    while True:
        print(n, end=' ')
        if n == 1:
            break
        if n % 2 == 1:
            n = 3*n + 1
        else:
            n = n // 2
    
if __name__ == '__main__':
    n = input()
    weird_algorithm(int(n))